app/assets/javascripts/modules/remove-filter.js

```diff
@@ -67,9 +67,10 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     }
 
     function getInput (removeFilterName, removeFilterValue, removeFilterFacet) {
-      var selector = (removeFilterName) ? " input[name='" + removeFilterName + "']" : " [value='" + removeFilterValue + "']"
+      var selector = (removeFilterName) ? "input[name='" + removeFilterName + "']" : "[value='" + removeFilterValue + "']"
+      var element = document.getElementById(removeFilterFacet)
 
-      return $('#' + removeFilterFacet).find(selector)
+      return element.querySelector(selector)
```

Comment:
Similar question here about our certainty of the elements existence. jQuery is very forgiving about elements not existing:

```
> jQuery('asdfasdf').find('adfasd').toArray()
[]
> document.getElementById('asdfasdf').querySelector('adfasd').toArray()
VM801:1 Uncaught TypeError: Cannot read property 'querySelector' of null
    at <anonymous>:1:36
```

Reply:
This caused me such a headache, couldn't work out why the tests were broken. As far as I know these will always exist, I've not been able to break it when running the app locally

Reply:
Thanks 👍 

I've been doing some reading of this to try form an understanding of what could go wrong. 

This `getInput` method strikes me as something that we should be prepared for the element not to exist. As it depends on another element existing in the facet. In fact I think I've found an example where it would fail: https://www.gov.uk/search/all?organisations%5B%5D=department-for-international-trade&order=updated-newest&parent=department-for-international-trade where there's an ability to filter on facets that aren't in the facet column (link originates from "See all latest documents" on [an organisation](https://www.gov.uk/government/organisations/department-for-international-trade)). 

I think you're ok on not adding a check to https://github.com/alphagov/finder-frontend/pull/2556/files#r657362674 as the element has to exist to get past this code (a nice optimisation would be to only look it up once)

I don't think GOV.UK has really formed a strategy for deciding when to care if elements exist or not (seems a fresh post jQuery problem) but to me it seems reasonable to assume elements exist if they're children of the component/module, and to always be cautious with anything beyond that. Does that make sense/sounds reasonable?

Reply:
So i think the input actually refers to the object that you're clicking on to remove. In the example that you posted `element.querySelector(selector)` finds `<input type="text" name="organisations[]" id="organisations_0" value="department-for-international-trade">` so it manages to find it even though there's no parent item on the page.

Reply:
We've spoken about this offline and I've updated the 1st commit message to reflect our findings